stage0/src/initramfs.rs

```diff
@@ -0,0 +1,55 @@
+//
+// Copyright 2023 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::fw_cfg::FwCfg;
+use core::slice;
+use x86_64::VirtAddr;
+
+pub struct RamDiskInfo {
+    pub address: VirtAddr,
+    pub size: u32,
+}
+
+/// Tries to load an initial RAM disk from the QEMU FW_CFG device.
+///
+/// If it finds a RAM disk it returns the address where it is loaded and the size. If not it
+/// returns `None`.
+pub fn try_load_initial_ram_disk(fw_cfg: &mut FwCfg) -> Option<RamDiskInfo> {
```

Comment:
Is there any value in having a RamDiskInfo struct instead of just returning `&'static [u8]` here?

Reply:
Good point. Done.